src/hooks/useTokenBalances.ts

```diff
@@ -83,28 +84,39 @@ async function _getBalances(walletInfo: WalletInfo, tokens: TokenDetails[]): Pro
   const contractAddress = depositApi.getContractAddress(networkId)
   assert(contractAddress, 'No valid contract address found. Stopping.')
 
-  const balancePromises: Promise<TokenBalanceDetails | null>[] = tokens.map((token) =>
-    fetchBalancesForToken(token, userAddress, contractAddress, networkId)
-      .then((balance) => {
+  const balancePromises: Promise<TokenBalanceDetails | null>[] = tokens.map((token) => {
+    const timeoutPromise = timeout<TokenBalanceDetails | null>({
```

Comment:
But this isn't `Promise<TokenBalanceDetails | null>`, this is `Promise<never>` as it will throw, always.